pypesto/sample/util.py

```diff
@@ -124,3 +127,252 @@ def bound_n_samples_from_env(n_samples: int):
     )
 
     return n_samples_new
+
+
+def harmonic_mean_log_evidence(
+    result: Result,
+    prior_samples: Optional[np.ndarray] = None,
+    neg_log_likelihood_fun: Optional[callable] = None,
+) -> float:
+    """
+    Compute the log evidence using the harmonic mean estimator.
+
+    Stabilized harmonic mean estimator is used if prior samples are provided.
+    Newton and Raftery (1994): https://doi.org/10.1111/j.2517-6161.1994.tb01956.x
+
+    Parameters
+    ----------
+    result: Result
+    prior_samples: np.ndarray (n_samples, n_parameters)
+        Samples from the prior distribution. If samples from the prior are provided,
+        the stabilized harmonic mean is computed (recommended). Then, the likelihood function must be provided as well.
+    neg_log_likelihood_fun: callable
+        Function to evaluate the negative log likelihood.
+    """
+    from scipy.optimize import minimize_scalar
+    from scipy.special import logsumexp
+
+    # compute negative log likelihood from traces
+    burn_in = geweke_test(result)
```

Comment:
you could check here if the result already has a burn-in (optionally, do this in the `geweke_test` function

Reply:
will add it in the `geweke_test`